### PR TITLE
VisualsManager follow up

### PR DESCRIPTION
**Pull request by daschuer**
This is a follow up to #1963, restoring 0.0vor various values after eject + some minor performance improvements,

### CODE REVIEW

File: src/engine/enginebuffer.cpp [hunk 1]
```diff
@@ -1179,11 +1180,6 @@ void EngineBuffer::postProcess(const int iBufferSize) {
 }
 
 void EngineBuffer::updateIndicators(double speed, int iBufferSize) {
-    if (!m_trackSampleRateOld || !m_trackSamplesOld) {
```
Comment (rryan) [inline src/engine/enginebuffer.cpp]: Wasn't this guarding against a divide by zero below?

Comment (daschuer) [reply src/engine/enginebuffer.cpp]: The function is not called in this case. But I have added it back as VERIFY_OR_DEBUG_ASSERT() as an extra savety net.

Comment (rryan) [reply src/engine/enginebuffer.cpp]: Ah I didn't notice, thanks.

**Comment by rryan** [comment]
LGTM

**Comment by daschuer** [comment]
Thank you for review

**Pull request merged**